web/META-INF/resources/js/controllers.js

```diff
@@ -0,0 +1,240 @@
+            $scope.hideoption = true;
+            var output = [];  
+            Autocomplete.get({term: string},
+                function success(response) {
+                    //alert($scope.challenge.question);
+                    //console.log("Success:" + JSON.stringify(response));
+                    //$scope.blogList = response;
+                    $scope.filterSearch = response;
+
+                },
+                function error(errorResponse) {
+                    console.log("Error:" + JSON.stringify(errorResponse));
+                }
+            );  
+        }  
+
+        $scope.fillTextbox = function(string){  
+            $scope.options.term = string;  
+            $scope.hidethis = true;  
+            $scope.hideoption = false;
+        }  
+
+        $scope.search = function(options) {
+            $rootScope.searchOptions = angular.copy(options);
+            
+            $location.path("/metadataView/"+options.term);
+
+        };
+
+    }]);
+    
+mudrodControllers.controller('vocabularyCtrl', ['$scope', '$rootScope', 'VocabList',
+    function vocabularyCtrl($scope, $rootScope, VocabList) {
+
+        var word = $rootScope.searchOptions.term;
+        var opt = $rootScope.searchOptions.preference;
+        VocabList.get({query: word},
+                function success(response) {
+                    //alert($scope.challenge.question);
+                    //console.log("Success:" + JSON.stringify(response));
+                    //$scope.blogList = response;
+                    $scope.ontologyList = response.graph.ontology;
+
+                },
+                function error(errorResponse) {
+                    console.log("Error:" + JSON.stringify(errorResponse));
+                }
+        );
+    }]);
+
+mudrodControllers.controller('metadataViewCtrl', ['$rootScope', '$scope', '$routeParams', 'MetaData', 'PagerService',
+    function metadataViewCtrl($rootScope, $scope, $routeParams, MetaData, PagerService) {
+        var vm = this;
+        vm.PDItems = [];
+        vm.pager = {};
+        vm.setPage = setPage;
+        vm.totalMatches = 0;
+
+        var word = new String();
+        var opt = new String();
+        
+        if(!$routeParams.name){
+            word = $rootScope.searchOptions.term;    
+            opt = $rootScope.searchOptions.preference; 
+        } else {
+            word = $routeParams.name;
+            var searchKeyWords = new String();
+
+            if (word.search(',') != -1) {
+                var topics = word.split(',');
+                for (var i = 0; i < topics.length; i++) {
+                    searchKeyWords += topics[i];
+                }
+            } else {
+                searchKeyWords = word;
+            }
+            
+            if (searchKeyWords.search('/') != -1 ) {
+                var topics = searchKeyWords.split('/');
+                searchKeyWords = "";
+                for (var i = 0; i < topics.length; i++) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Rename "i" as this name is already used in declaration at line 100. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.com/coding_rules#rule_key=javascript%3AS2814)